flow/layers/container_layer.cc

```diff
@@ -6,13 +6,42 @@
 
 namespace flutter {
 
-ContainerLayer::ContainerLayer() {}
+ContainerLayer::ContainerLayer()
+    : child_needs_screen_readback_(false), renders_to_save_layer_(false) {}
 
 ContainerLayer::~ContainerLayer() = default;
 
 void ContainerLayer::Add(std::shared_ptr<Layer> layer) {
-  layer->set_parent(this);
+  Layer* the_layer = layer.get();
```

Comment:
I feel all the `the_layer` in this function can be replaced with `layer` so we don't actually need `the_layer`? The last `the_layer` can be replaced with `layer.get()`. But as I'll comment later, I think sending in a `const Layer&` is probably a little safer than sending in a `Layer*`  to `UpdateChildReadback`.

Reply:
Except that the last one happens after an std::move() and so layer.get() is invalid. And, it can't be moved before the layers_.add() because a precondition for that use is that the layer must already be a child. I'd have to use layers_.last.get() which felt awkward.

Reply:
I've left this alone due to the issue of needing the value of layer.get() after it has been std::moved...